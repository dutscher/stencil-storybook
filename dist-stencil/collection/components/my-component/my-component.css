/* stylelint-disable scale-unlimited/declaration-strict-value */
@font-face {
  font-family: "Roboto";
  src: url("../fonts/Roboto/light.woff") format("woff");
  font-weight: 400;
  font-style: normal;
  font-display: swap;
}
@font-face {
  font-family: "Roboto";
  src: url("../fonts/Roboto/regular.woff") format("woff");
  font-weight: 500;
  font-style: normal;
  font-display: swap;
}
@font-face {
  font-family: "Roboto";
  src: url("../fonts/Roboto/bold.woff") format("woff");
  font-weight: 700;
  font-style: normal;
  font-display: swap;
}
:host {
  display: block;
  font-family: sans-serif;
}

.my-component {
  font-size: 16px;
  display: block;
}
.my-component__name {
  color: #fba308;
}
.my-component__name .my-component__example-for-inner-block {
  color: #00abba;
}
.my-component--modified {
  text-align: center;
}