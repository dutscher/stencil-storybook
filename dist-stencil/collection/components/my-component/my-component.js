var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
  var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
  if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
  else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
  return c > 3 && r && Object.defineProperty(target, key, r), r;
};
import { Component, Prop, Event, Host, h } from '@stencil/core';
import { format } from '../../utils/utils';
import { BEM, hostClasses, ComponentName } from '../../utils';
export class MyComponent {
  constructor() {
    /** Modifier Prop */
    this.modifier = '';
  }
  getText() {
    return format(this.first, this.middle, this.last);
  }
  render() {
    const cssClasses = hostClasses({
      componentName: this.componentName,
      modifier: this.modifier,
    });
    return (h(Host, { class: `${cssClasses} dasdsa---dddsd container`, onClick: () => this.removeButtonResponse.emit({ test: 'test' }) },
      "Hello, World! I'm ",
      h("span", { class: BEM(this, 'name') },
        " ",
        this.getText()),
      h("slot", null)));
  }
  static get is() { return "my-component"; }
  static get originalStyleUrls() { return {
    "$": ["my-component.scss"]
  }; }
  static get styleUrls() { return {
    "$": ["my-component.css"]
  }; }
  static get properties() { return {
    "first": {
      "type": "string",
      "mutable": false,
      "complexType": {
        "original": "string",
        "resolved": "string",
        "references": {}
      },
      "required": false,
      "optional": false,
      "docs": {
        "tags": [],
        "text": "First Prop"
      },
      "attribute": "first",
      "reflect": false
    },
    "middle": {
      "type": "string",
      "mutable": false,
      "complexType": {
        "original": "string",
        "resolved": "string",
        "references": {}
      },
      "required": false,
      "optional": false,
      "docs": {
        "tags": [],
        "text": "Middle Prop"
      },
      "attribute": "middle",
      "reflect": false
    },
    "last": {
      "type": "string",
      "mutable": false,
      "complexType": {
        "original": "string",
        "resolved": "string",
        "references": {}
      },
      "required": false,
      "optional": false,
      "docs": {
        "tags": [],
        "text": "Last Prop"
      },
      "attribute": "last",
      "reflect": false
    },
    "modifier": {
      "type": "string",
      "mutable": false,
      "complexType": {
        "original": "string",
        "resolved": "string",
        "references": {}
      },
      "required": false,
      "optional": false,
      "docs": {
        "tags": [],
        "text": "Modifier Prop"
      },
      "attribute": "modifier",
      "reflect": false,
      "defaultValue": "''"
    }
  }; }
  static get events() { return [{
      "method": "removeButtonResponse",
      "name": "removeButtonResponse",
      "bubbles": true,
      "cancelable": true,
      "composed": true,
      "docs": {
        "tags": [],
        "text": "Prevent"
      },
      "complexType": {
        "original": "any",
        "resolved": "any",
        "references": {}
      }
    }]; }
}
__decorate([
  ComponentName()
], MyComponent.prototype, "componentName", void 0);
