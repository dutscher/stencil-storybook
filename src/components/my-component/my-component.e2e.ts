import { newE2EPage } from '@stencil/core/testing';
import { getComponentNameFromFilename, setupE2E } from '../../utils/tests/setup';

const componentName = getComponentNameFromFilename(__filename);

describe(componentName, () => {
  it('default', async () => {
    const { page, component } = await setupE2E(componentName, null);
    await page.waitForChanges();
    expect(component).toMatchSnapshot();
  });

  it('default manually', async () => {
    const page = await newE2EPage();

    await page.setContent('<my-component></my-component>');
    const element = await page.find(componentName);
    expect(element).toHaveClass('hydrated');
  });

  it('renders changes to the name data', async () => {
    const page = await newE2EPage();

    await page.setContent('<my-component></my-component>');
    const component = await page.find(componentName);
    const element = await page.find(componentName + ' >>> div');
    expect(element.textContent).toEqual(`Hello, World! I'm `);

    component.setProperty('first', 'James');
    await page.waitForChanges();
    expect(element.textContent).toEqual(`Hello, World! I'm James`);

    component.setProperty('last', 'Quincy');
    await page.waitForChanges();
    expect(element.textContent).toEqual(`Hello, World! I'm James Quincy`);

    component.setProperty('middle', 'Earl');
    await page.waitForChanges();
    expect(element.textContent).toEqual(`Hello, World! I'm James Earl Quincy`);
  });
});
