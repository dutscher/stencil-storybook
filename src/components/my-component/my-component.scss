// with shadow dom
:host {
  display: block;
  font-family: sans-serif;
}
$selector: '.my-component';

#{$selector} {
  font-size: ms(0);
  display: block;
  // block
  &__name {
    color: $color-primary;

    #{$selector}__example-for-inner-block {

    }
  }
  // modifier
  &--modified {
    text-align: center;
  }
}
