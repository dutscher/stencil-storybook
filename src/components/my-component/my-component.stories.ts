import { html } from 'lit-html';
import markdown from './readme.md';
import { eventHandles, action } from '../../../.storybook/helpers/custom-action';

const costumEvents = ['clicked'];
const events = ['mouseover', 'click', ...eventHandles(costumEvents)];

const args = {
  modifier: '',
  first: 'First',
  middle: 'Middle',
  last: 'Last',
  hex: '#fba308',
  boolean: false,
  radio: '',
  select: '',
};
// https://storybook.js.org/docs/react/essentials/controls#annotation
const argTypes = {
  modifier: {
    control: {
      type: 'select',
      options: ['', 'modified'],
    },
  },
  first: {
    description: 'The first name',
  },
  middle: {
    description: 'The middle name',
  },
  last: {
    description: 'The last name',
  },
  hex: {
    control: 'color',
  },
  radio: {
    control: {
      type: 'inline-radio',
      options: ['loading', 'error', 'ready'],
    },
  },
};

// https://storybook.js.org/docs/react/writing-stories/parameters
export default {
  title: 'Components/My Component',
  component: 'my-component',
  parameters: {
    notes: {markdown},
    actions: {
      handles: events,
    },
  },
};

const Template = (
  {
    modifier,
    first,
    middle,
    last,
    hex,
  }) => html`
  <my-component
    style="color:${hex}"
    modifier="${modifier}"
    first="${first}"
    middle="${middle}"
    last="${last}">
  </my-component>
  ${action('my-component', costumEvents)}
`;

export const Default = Template.bind({});

Default.args = {...args};
Default.argTypes = {...argTypes};

export const Modified = Template.bind({});
Modified.args = {...args, modifier: 'modified'};
Modified.argTypes = {...argTypes};
