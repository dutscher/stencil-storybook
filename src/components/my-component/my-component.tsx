import { Component, Prop, Event, EventEmitter, Host, h } from '@stencil/core';
import { format } from '../../utils/utils';
import { BEM, hostClasses } from '../../utils';

@Component({
  tag: 'my-component',
  styleUrl: 'my-component.scss',
  shadow: false,
})
export class MyComponent {
  private readonly componentName: string = 'my-component';

  @Prop() first: string;

  @Prop() middle: string;

  @Prop() last: string;

  @Prop() modifier: string = '';

  @Event({bubbles: false}) clicked!: EventEmitter<void>;

  private getText(): string {
    return format(this.first, this.middle, this.last);
  }

  render() {
    const cssClasses = hostClasses({
      componentName: this.componentName,
      modifier: this.modifier,
    });

    return (
      <Host
        class={cssClasses}
        onClick={() => this.clicked.emit()}>
        Hello, World! I'm <span class={BEM(this, 'name')}> {this.getText()}</span>
      </Host>
    );
  }
}
