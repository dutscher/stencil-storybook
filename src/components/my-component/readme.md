# my-component

<!-- Auto Generated Below -->
