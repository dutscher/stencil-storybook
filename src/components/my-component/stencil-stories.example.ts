// this file is generated by 'stencil-to-storybook.js' aka pre-storybook npm task
// and it is git to show this whats happen in github ;)

import markdown from './readme.md';
import { a, h } from '../../utils/storybook';

export default {
  ...{
  "component": "my-component",
  "args": {
    "first": "",
    "middle": "",
    "last": "",
    "modifier": "",
    "maxFileSize": "",
    "multilineShizzle": false
  },
  "argTypes": {
    "first": {
      "description": "First Prop"
    },
    "middle": {
      "description": "Middle Prop"
    },
    "last": {
      "description": "Last Prop"
    },
    "modifier": {
      "description": "Modifier Prop"
    },
    "maxFileSize": {
      "description": "Specify size limit in Kb"
    },
    "multilineShizzle": {
      "description": "Yo im a multiline\ncan you see me?"
    }
  },
  "customEvents": [
    "removeButtonResponse"
  ],
  "slots": {
    "defaultSlot": ""
  }
},
  markdown,
  tpl: ({ id, first, middle, last, modifier, maxFileSize, multilineShizzle, slot }) => h`
    <my-component
      id="${a(id)}"
			first="${a(first)}"
			middle="${a(middle)}"
			last="${a(last)}"
			modifier="${a(modifier)}"
			maxFileSize="${a(maxFileSize)}"
			multilineShizzle="${a(multilineShizzle)}"
    >
		${slot}
	</my-component>`
};
