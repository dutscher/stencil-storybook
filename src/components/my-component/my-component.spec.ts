import { newSpecPage } from '@stencil/core/testing';
import { setupSpec, getComponentNameFromFilename } from '../../utils/tests/setup';
import { MyComponent } from './my-component';

const componentName = getComponentNameFromFilename(__filename);

describe(componentName, () => {
  it('default', async () => {
    const { component } = await setupSpec(MyComponent);
    expect(component.outerHTML).toMatchSnapshot();
  });

  it('vs. default manually', async () => {
    const { root } = await newSpecPage({
      components: [MyComponent],
      html: `<${componentName}></${componentName}>`,
    });
    expect(root).toMatchSnapshot();
  });
});
