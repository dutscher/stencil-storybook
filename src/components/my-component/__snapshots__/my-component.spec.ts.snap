// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`my-component default 1`] = `"<my-component class=\\"my-component dasdsa---dddsd container\\"><!----> Hello, World! I'm <span class=\\"my-component__name\\"></span>  </my-component>"`;

exports[`my-component vs. default manually 1`] = `
<my-component class="container dasdsa---dddsd my-component">
  <!---->
  Hello, World! I'm
  <span class="my-component__name"></span>
</my-component>
`;
